drivers/ethernet/eth_native_posix.c

```diff
@@ -321,6 +322,8 @@ static int read_data(struct eth_context *ctx, int fd)
 		count += frag->len;
 	} while (ret > 0);
 
+	type = NET_ETH_HDR(pkt)->type;
```

Comment:
I think that's something we'll need to think about. It piles up more stuff to add in every driver, while it could be centralized. I still would like to see this put into ethernet l2 somehow, here at this point you don't know if the packet is meant for a raw socket or not (and this approach would also make the union stuff as seen in previous patch impossible).